src/compas/geometry/primitives/frame.py

```diff
@@ -180,6 +180,8 @@ def __iter__(self):
         return iter([self.point, self.xaxis, self.yaxis])
 
     def __eq__(self, other, tol=1e-05):
+        if not hasattr(other, '__iter__') or not hasattr(other, '__len__') or len(self) != len(other):
+            return False
```

Comment:
aren't `__iter__` and `__len__` automatically covered by `zip`?
also, you could use `zip_longest` instead of `zip` which will produce `None` for one of the iterables if the length is not the same...

Reply:
Yes and no.  If `other` doesn't have `__iter__`, `zip` will raise an error.  I feel that `__eq__` should never raise an error. For example `[0, 0, 0] == 0` would return `False` rather than raise an error, and if compas primitives are supposed to behave like python primitives, then we should mimic that. 

Using `zip_longest` would also throw an exception in the case you describe because you can't perform arithmetic on `None`.

Reply:
okay. fair enough...